### ISSUE TITLE
Nextion button issues

### ISSUE THREAD

**Issue by ANTI443** [klotzsch-lab/Maasi#1]
Hello, great project.

My issue is the Nextion display. The touch works, when I try to touch + positive or - negative nothing happens. They display indicates I am interacting but nothing. Also the start button doesn't start I can go into the menu only and back.

Any suggestions what to look at?

Cheers

**Comment by dani-carbonell** [comment]
Hi Annti443,

Thanks for the interest in the project.

Have you flashed the ESP32?
Currently the Nextion display only reports the buttons pressed and it is the ESP32 witch sends back the new value to ve displayed.
Do you have the ESP32 connected to the Nextion display according to the diagrams?

**Comment by ANTI443** [comment]
> Hi Annti443,
> 
> Thanks for the interest in the project.
> 
> Have you flashed the ESP32?
> Currently the Nextion display only reports the buttons pressed and it is the ESP32 witch sends back the new value to ve displayed.
> Do you have the ESP32 connected to the Nextion display according to the diagrams?



Hello, Dani thank you for the reply Yes, I have flashed 4 ESP32 with no errors. I also used the same settings as you did in your YouTube video shown below.

![Screenshot_20220318-142550_YouTube.jpg](https://user-images.githubusercontent.com/61336792/159071014-44eb4d43-ae75-4637-a55f-832b6f72dabd.jpg)


Are the GPIO pins correct? 
 
Cheers!

**Comment by ANTI443** [comment]
> Hi Annti443,
> 
> Thanks for the interest in the project.
> 
> Have you flashed the ESP32?
> Currently the Nextion display only reports the buttons pressed and it is the ESP32 witch sends back the new value to ve displayed.
> Do you have the ESP32 connected to the Nextion display according to the diagrams?

To Add I also tried this on 3 different brand ESP32 modules to see if it is my microcontrollers. With no success

**Comment by dani-carbonell** [comment]
I currently haven't access to de prototype to test the code. I will be able to test it by the end of the week.
The issue might be related to the Serial configuration of the [Nextion library](https://github.com/klotzsch-lab/Maasi/blob/main/SOFTWARE/Arduino/ITEADLIB_Arduino_Nextion/NexConfig.h).

**Comment by ANTI443** [comment]
> I currently haven't access to de prototype to test the code. I will be able to test it by the end of the week. The issue might be related to the Serial configuration of the [Nextion library](https://github.com/klotzsch-lab/Maasi/blob/main/SOFTWARE/Arduino/ITEADLIB_Arduino_Nextion/NexConfig.h).

Yes please, I have been looking at the code. Everything seems generally good, I'm not a expert.

What I have done:
1. Re- installed arduino on my PC and installed it on 2 other computers to double check. 
2. I have installed the exact arduino version which was 1.8.16 If I recall.
3. I have installed the same versions of libraries and board managers 
4. I made sure my BLHELI32 is installed correctly with firmware version.
5. i have used different esp32 boards from different makers 
6. Everything complies correctly

**Comment by ANTI443** [comment]
@dani-carbonell 

Hey Dani, any luck on getting the original, Or comparing the 2?

Thank you 😁!

**Comment by ANTI443** [comment]
@dani-carbonell 

hey Dani any luck on looking at the Source code for Maasi? I have been trying to debug the serial interface like you said.

Any thoughts, I still can't figure out the issue.
Me and a few others are trying to replicate your project.

Thank you 😁

**Comment by dani-carbonell** [comment]
Hi ANTTI443,
I finally gained access to a prototype again and was able to reproduce the error. Unfortunately I am currently travelling and was not able to find out what the cause is. I hope that by the end of this month I have a solution for this issue. I am sorry for the delays, and thanks again for keeping the interest in the project.

**Comment by dani-carbonell** [comment]
As a temporary solution I suggest:

Using code in commit 9756e8dd67690d672f7acaa532320a6e751e70f8

Commenting line https://github.com/klotzsch-lab/Maasi/blob/9756e8dd67690d672f7acaa532320a6e751e70f8/SOFTWARE/Arduino/maasi/maasi.ino#L286

**Comment by ANTI443** [comment]
> Hi ANTTI443,
> I finally gained access to a prototype again and was able to reproduce the error. Unfortunately I am currently travelling and was not able to find out what the cause is. I hope that by the end of this month I have a solution for this issue. I am sorry for the delays, and thanks again for keeping the interest in the project. 

@dani-carbonell thank you very much for you're amazing research tool you created.

I greatly appreciate your help and support.

We are all greatful for your continuous help,  safe travels!!! I will look at your suggestions.
😁

**Comment by ANTI443** [comment]
@dani-carbonell I fix the issue!


Steps I did
1. I downgraded arduino 
2. I used a different version of blheli32 32.6 vs 32.7

3. I changed the esp32 version to the one you recommended.
4. I changed the baud rate in tools to 115200.

I can't confirm those steps make any difference, when with many trial and error these settings ended up working.

Elapsedmillis also has a different name of author now, same people when you go to the web link.

Hope this helps anyone.

**Comment by ANTI443** [comment]
Additional I am using a esp32 with esp wroom 32. This board is the same as in the diagrams with yellow pins. I tried different esp32 boards but had issues for some reason, . Could be the boards are designed slightly different, pins are different or could be me.
